Chart: Implement `controller.admissionWebhooks.service.servicePort`.

ralgozino opened the pull request:
<!--- Provide a general summary of your changes in the Title above --->
<!--- Please don't @-mention people in PR or commit messages (do so in an additional comment). --->
<!--- Please make sure you title is descriptive, it is used in the Release notes to let others know what it does ---> 

## What this PR does / why we need it:
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

While the chart values.yaml provides a controller.admissionWebhooks.service.servicePort option, its value was being ignored by the chart template.

This PR changes the template to use the value for the admission webhook service port in the Service definition and in the corresponding ValidatingWebhookConfiguration instead of ignoring it.

The PR also adds some Helm unit tests for this value.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] CVE Report (Scanner found CVE and adding report)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation only

## Which issue/s this PR fixes
<!--
(optional, in `fixes #<issue number>` format, will close that issue when PR gets merged):

fixes #
-->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- Manually tested that the change works as expected (service uses the custom port, webhook configuration point to the right port).
- Launched `make kind-e2e-chart-tests` locally (mac M1): all tests passed
- Launched `helm unittest charts/ingress-nginx -d` locally (mac M1): all tests passed.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [x] I have added unit and/or e2e tests to cover my changes.
- [x] All new and existing tests passed.

Gacko commented:
/triage accepted
/kind bug
/priority backlog
/hold

Thanks for your contribution! I'll review it later, hopefully! 🙂

ralgozino commented:
thank you @Gacko 

This is my first contribution to the project, please let me know if I missed anything and I'll happily fix it :-) 

Have a nice day!

Gacko commented:
/ok-to-test
/lgtm
/unhold

Gacko commented:
Ciao!

Since you enabled the option for maintainers to edit your fork, I implemented a few improvements, mainly around the tests, and rebased your branch.

Mille grazie e buona serata!

Gacko commented:
/cherry-pick release-1.10

Gacko commented:
/cherry-pick release-1.11

ralgozino commented:
> Ciao!
> 
> Since you enabled the option for maintainers to edit your fork, I implemented a few improvements, mainly around the tests, and rebased your branch. Apart from that, your proposal looks good and is ready to be merged.
> 
> Mille grazie e buona serata!

Thank you for the feedback. I saw your changes, they make a lot of sense.

Buona giornata and see you in the next PR :-)
